packages/components/src/utils/unit-values.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { isNil } from 'lodash';
+
+const UNITED_VALUE_REGEX = /^([\d.\-+]*)\s*(fr|cm|mm|Q|in|pc|pt|px|em|ex|ch|rem|lh|vw|vh|vmin|vmax|%|cap|ic|rlh|vi|vb|deg|grad|rad|turn|s|ms|Hz|kHz|dpi|dpcm|dppx)?$/;
+
+/**
+ * Parses a number and unit from a value.
+ *
+ * @param  toParse Value to parse
+ *
+ * @return  The extracted number and unit.
+ */
+export function parseCSSUnitValue(
+	toParse: string | number
+): [ number | undefined, string | undefined ] {
```

Comment:
Since we can't return `[undefined, string]`, should we be more precise on the return type? Something like

`type ParsedUnitValue = [number, string | undefined] | [undefined, undefined]`

Reply:
Sure.

Reply:
Sure, feel free to ignore this comment